src/Aspire.Hosting/Lifecycle/IDistributedApplicationLifecycleHook.cs

```diff
@@ -20,4 +20,14 @@ Task BeforeStartAsync(DistributedApplicationModel appModel, CancellationToken ca
     {
         return Task.CompletedTask;
     }
+
+    /// <summary>
+    /// Executes after the distributed application stops.
+    /// </summary>
+    /// <param name="cancellationToken">The cancellation token.</param>
+    /// <returns>A <see cref="Task"/> representing the asynchronous operation.</returns>
+    Task AfterStopAsync(CancellationToken cancellationToken = default)
```

Comment:
@mitchdenny

Reply:
While the current approach works, it requires pulling the hooks out of the DI container before the stop actually happens, as otherwise the DI container is no longer available. This could be an issue if hooks rely on the DI container disposal process.  I'm not sure if there's a better point where the stop handlers could be invoked *after* DCP has brought down everything but *before* the application host has been torn down.

Reply:
Hmm, we should resolve that. It might be possible to just clean up in dispose instead.

Reply:
Switched to using `IDisposable` on the hook instead, which is cleaner and seems to work well enough for now.

Reply:
Nice!